docs/saved_objects/saved_object_access_control_design_proposal.md

```diff
@@ -0,0 +1,216 @@
+
+### Workspace and Saved Objects
+
+Workspace is a newly introduced type of OpenSearch Dashboards data, it is a container which organize the functions and saved objects, so the saved object in a workspace inherit the permissions settings from the workspace. One saved object can potentially in multiple workspaces, thus it inherits the permission settings from all the workspaces it is assigned to.
+
+#### Permissions
+
+Based on the proposed workspace experience, an user may have following type of permissions on OpenSearch Dashboards data:
+
+* read
+* write (read_and_write)
+    * the write permission means the user update the saved object, and can add this saved object to another workspace, or update the ACL of the object so that the object can be shared to others
+
+For workspaces, a user may have following 3 type of permissions:
+
+* management
+* library_read
+* library_write (library_read_and_write)
+
+the management permission allows the principal to update the workspace settings, like enabling or disabling features in the workspace, as well as updating the ACLs for the workspace, such as granting permissions of other principals to access this workspace. *library_read* and *library_write* permission will be used to control the user/group permissions on the saved objects in the workspace.
+
+With this permission definition
+
+* a user is the workspace admin if they have management permission of the workspace
+* a user can access/use a workspace if they have either library_read, or library_write permission or management permission of the workspace
+* a user can view a saved object if 
+    * they have read or write permission of the saved object, or 
+    * they have library_read or library_write or admin permission of one of the workspace that saved object is associated to.
+* a user can edit a saved object if
+    * they have write permission of the saved object, or
+    * they have library_write or admin permission of one of the workspace that saved object is associated to
+* a user can create saved object in a workspace if they have library_write permission of the workspace
+
+
+#### ACL
+
+The ACLs can be represented as following:
+
+```
+{
+  "permissions": {
+    "<permission_type_1>": [
+       "<principal_1>",
+       "<principal_2>"
+    ],
+    "<permission_type_2>": [
+       "<principal_3>",
+       "<principal_4>"
+    ],
+  } 
+}
+```
+
+We can also allow wildcard `*` to indicate any authenticated identity in the ACL.
+
+For example, if we want to allow finance_manager group to manage the workspace, and allow *finance_analyst* groups to build dashboards in the workspace, the workspace ACLs may look like:
+
+```
+{
+  "permissions": {
+    "management": [
+       "group/finance_manager"
```

Comment:
Is there a system that will translate this into a principal/set of principals?  Or is this assumed to be a principal part of the AuthN system?